File: 7_stacks_and_queues/brackets.py
def solution(A):
    lefts = ['(', '{', '[']
    rights =  [')', '}', ']']

    stack = []

    for i in A:
        if i in lefts:
            stack.append(i)
        else:
            j = rights.index(i)
            try:
                now = stack.pop()
            except:
                return 0
            if now != lefts[j]:
                return 0
    if len(stack) == 0:
        return 1
    else:
        return 0

File: 07_stacks_and_queues/brackets.cpp
#include "string"
#include "stack"

using namespace std;

int solution(string &s) {
  int n = s.length();
  stack<char> stk;
  for (int i = 0; i < n; i++) {
    if (stk.size() > 0 && stk.top() == s[i]) {
      stk.pop();
    }
    else if (s[i] == '(') {
      stk.push(')');
    }
    else if (s[i] == '{') {
      stk.push('}');
    }
    else if (s[i] == '[') {
      stk.push(']');
    }
    else {
      return 0;
    }
  }
  if (stk.size() == 0) {
    return 1;
  }
  else {
    return 0;
  }
}
